src/Symfony/Component/Serializer/Mapping/AttributeMetadata.php

```diff
@@ -127,17 +127,45 @@ public function getMaxDepth(): ?int
     /**
      * {@inheritdoc}
      */
-    public function setSerializedName(string $serializedName = null)
+    public function setSerializedNames(array $serializedNames): void
     {
-        $this->serializedName = $serializedName;
+        $this->serializedNames = $serializedNames;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getSerializedName(): ?string
+    public function addSerializedName(?string $serializedName, array $groups): void
     {
-        return $this->serializedName;
+        if (empty($groups)) {
+            $this->serializedNames['without-group'] = $serializedName;
```

Comment:
```suggestion
            $this->serializedNames['*'] = $serializedName;
```
To be consistent with `setNormalizationContextForGroups`?